### ISSUE TITLE
Unable to install it on Macbook m1 

### ISSUE THREAD

**Issue by kiranchavala** [AlmirKadric-Published/docker-tuntap-osx#42]
Not able to install tuntap

 brew cask install tuntap

==> Installing Cask tuntap
==> Running installer for tuntap; your password may be necessary.
Package installers may write to any location; options such as `--appdir` are ignored.
Password:
installer: Package name is TunTap Installer package
installer: Installing at base path /
installer: The install failed. (The Installer encountered an error that caused the installation to fail. Contact the software manufacturer for assistance. An error occurred while running scripts from the package “tuntap_20150118.pkg”.)
==> Purging files for version 20150118 of Cask tuntap
Error: Failure while executing; `/usr/bin/sudo -E -- /usr/bin/env LOGNAME=kiran_chavala USER=kiran_chavala USERNAME=kiran_chavala /usr/sbin/installer -pkg /opt/homebrew/Caskroom/tuntap/20150118/tuntap_20150118.pkg -target /` exited with 1. Here's the output:
installer: Package name is TunTap Installer package
installer: Installing at base path /
installer: The install failed. (The Installer encountered an error that caused the installation to fail. Contact the software manufacturer for assistance. An error occurred while running scripts from the package “tuntap_20150118.pkg”.)

**Comment by kiranchavala** [comment]
Seems like a open issue at  https://github.com/Tunnelblick/Tunnelblick/issues/682

**Comment by kiranchavala** [comment]
https://github.com/Tunnelblick/Tunnelblick/issues/682

**Comment by vinobc** [comment]
same for me as well

**Comment by gitrojones** [comment]
Incase this helps anyone else:

A good alternative has been setting up an ovpn container to run in docker and using Tunnelblick to connect with our needed CIDR ranges. Not as ideal as a proper host bridge but its pretty close.

**Comment by roupasz** [comment]
Hi @gitrojones , I stumbled upon this issue and I was wondering if you can you please elaborate with specific steps on how i could use the ovpn container/Tunnelblick setup that you are referring to?
I would like to be able to create bridge interface between the Physical Machine and the Host Virtual Machine which isn't possible currently as tuntap is not supported in M1 and from my understanding this can be achieved with your approach?
Thanks in advance

**Comment by gitrojones** [comment]
> Hi @gitrojones , I stumbled upon this issue and I was wondering if you can you please elaborate with specific steps on how i could use the ovpn container/Tunnelblick setup that you are referring to? I would like to be able to create bridge interface between the Physical Machine and the Host Virtual Machine which isn't possible currently as tuntap is not supported in M1 and from my understanding this can be achieved with your approach? Thanks in advance

Yep works well with M1. Specific steps depends on your setup but essentially you'd just bring up OVPN in docker via something like [https://hub.docker.com/r/kylemanna/openvpn/](https://hub.docker.com/r/kylemanna/openvpn/), copy the generated certificate to your local machine then connect to the above container using an OVPN client like Tunnelblick via the certificate. You should be able to follow nearly step-by-step with the steps on the docker image above.

I'm not working on OSX systems anymore but we had a handful of dev's w/ M1 machines using this approach without issue.

**Comment by roupasz** [comment]
> > Hi @gitrojones , I stumbled upon this issue and I was wondering if you can you please elaborate with specific steps on how i could use the ovpn container/Tunnelblick setup that you are referring to? I would like to be able to create bridge interface between the Physical Machine and the Host Virtual Machine which isn't possible currently as tuntap is not supported in M1 and from my understanding this can be achieved with your approach? Thanks in advance
> 
> Yep works well with M1. Specific steps depends on your setup but essentially you'd just bring up OVPN in docker via something like https://hub.docker.com/r/kylemanna/openvpn/, copy the generated certificate to your local machine then connect to the above container using an OVPN client like Tunnelblick via the certificate. You should be able to follow nearly step-by-step with the steps on the docker image above.
> 
> I'm not working on OSX systems anymore but we had a handful of dev's w/ M1 machines using this approach without issue.

Thanks a lot for the info and time, i will check the guide and follow up the openvpn steps.

**Comment by roupasz** [comment]
> > Hi @gitrojones , I stumbled upon this issue and I was wondering if you can you please elaborate with specific steps on how i could use the ovpn container/Tunnelblick setup that you are referring to? I would like to be able to create bridge interface between the Physical Machine and the Host Virtual Machine which isn't possible currently as tuntap is not supported in M1 and from my understanding this can be achieved with your approach? Thanks in advance
> 
> Yep works well with M1. Specific steps depends on your setup but essentially you'd just bring up OVPN in docker via something like https://hub.docker.com/r/kylemanna/openvpn/, copy the generated certificate to your local machine then connect to the above container using an OVPN client like Tunnelblick via the certificate. You should be able to follow nearly step-by-step with the steps on the docker image above.
> 
> I'm not working on OSX systems anymore but we had a handful of dev's w/ M1 machines using this approach without issue.

Thanks a lot for the info and time, i will check the guide and follow up the openvpn steps.
